scripts/inference/ft/mpt_ckpt_to_ft.py

```
@@ -0,0 +1,294 @@
+            split_vals[j].tofile(save_path)
+            if config['no_bias']:
+                write_empty_bias(key, save_path, (3, split_vals[j].shape[-1]))
+
+    else:
+        print("[ERROR] cannot find key '{}'".format(key))
+
+
+def split_and_convert(args):
+    save_dir = args.save_dir + '/%d-gpu/' % args.infer_gpu_num
+
+    if (os.path.exists(save_dir) == False):
+        os.makedirs(save_dir)
+
+    factor = args.infer_gpu_num
+
+    # do conversion on cpu
+    torch_device = 'cpu'
+
+    model = transformers.AutoModelForCausalLM.from_pretrained(
+        args.name_or_dir, trust_remote_code=True).to(torch_device)
+
+    hf_config = vars(model.config)
+
+    config = configparser.ConfigParser()
+    config['gpt'] = {}
+    try:
+        config['gpt']['model_name'] = 'mpt' if hf_config[
+            '_name_or_path'] == '' else hf_config['_name_or_path']
+        config['gpt']['head_num'] = str(hf_config['n_heads'])
+        n_embd = hf_config['d_model']
+        config['gpt']['size_per_head'] = str(n_embd // hf_config['n_heads'])
+        config['gpt']['inter_size'] = str(n_embd * hf_config['expansion_ratio'])
+        config['gpt']['max_pos_seq_len'] = str(hf_config['max_seq_len'])
+        config['gpt']['num_layer'] = str(hf_config['n_layers'])
+        config['gpt']['vocab_size'] = str(hf_config['vocab_size'])
+        config['gpt']['start_id'] = str(hf_config['bos_token_id'])
+        config['gpt']['end_id'] = str(hf_config['eos_token_id'])
+        config['gpt']['weight_data_type'] = args.weight_data_type
+        config['gpt']['tensor_para_size'] = str(args.infer_gpu_num)
+        if hf_config['attn_config']['alibi']:
+            config['gpt']['has_positional_encoding'] = str(False)
+            config['gpt']['use_attention_linear_bias'] = str(True)
+
+        with open(save_dir + '/config.ini', 'w') as configfile:
+            config.write(configfile)
+    except:
+        print(f'Fail to save the config in config.ini.')
```

can we raise the actual exception here?

---

sure